### PR TITLE
Improved implementation of the `expand` operator

### PR DESCRIPTION
**Pull request by antvaset**
According to the spec for the `expand` operator:

> If the interval boundaries are more precise than the per quantity, the more precise values will be truncated to the precision specified by the per quantity.

This PR improves the current implementation but I'll need to look into it further and fully test all potential edge cases.

### CODE REVIEW

**Comment by antvaset** [comment]
This is failing the CqlMainSuiteTest. In draft state until fixed.

**Review by JPercival — APPROVED** [review]

File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/runtime/DateTimeTest.java [hunk 1]
```diff
@@ -368,4 +368,74 @@ void dateTimeEquals() {
 
         // TODO: It'd be good to extend these across different types of constructors, for example, from a string
     }
+
+    @Test
+    void roundToPrecision() {
```
Comment (lukedegruchy) [inline Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/runtime/DateTimeTest.java:373]: Suggestion:  try with different timezones and bonus points for a weird half-hour timezone, like America/St_Johns

**Review by lukedegruchy — APPROVED** [review]
